src/edit.js

```diff
@@ -36,46 +48,66 @@ export default function Edit( { attributes, setAttributes } ) {
 	useEffect( () => {
 		async function getRelatedPosts() {
 			const postId = window.location.href.match( /post=([\d]+)/ )[ 1 ];
```

Comment:
The `postId` can be received from the [context](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-context/) or, alternatively, using `useSelect()` hook (`select( 'core/editor' ).getCurrentPostId()`).

Reply:
@delawski neat! I did not think about that.